toplevel/opttoploop.ml

```diff
@@ -682,3 +699,23 @@ let run_script ppf name args =
     else name
   in
   use_silently ppf explicit_name
+
+let preprocess_phrase_with_log = preprocess_phrase
+
+let preprocess_phrase ppf phr =
+  let log = Location.init_log ppf in
+  let ans = preprocess_phrase log phr in
```

Comment:
it would be slightly nicer to avoid the shadowing here with another name for the inner `preprocess_phrase` function?